R/bayesiannetworkanalysis.R

```diff
@@ -193,67 +202,146 @@ BayesianNetworkAnalysis <- function(jaspResults, dataset, options) {
-                                       algorithm  = "rjmcmc",
-                                       iter       = as.numeric(options[["iter"]]),
-                                       save       = TRUE,
-                                       burnin     = as.numeric(options[["burnin"]]),
-                                       g.start    = options[["initialConfiguration"]],
-                                       df.prior   = as.numeric(options[["dfprior"]]),
-                                       g.prior    = as.numeric(options[["gprior"]])))
-
-    if (isTryError(bdgraphFit)) {
-      message <- .extractErrorMessage(bdgraphFit)
-      if (startsWith(message, "system is computationally singular")) {
-        .quitAnalysis(gettext("Could not invert the posterior rate matrix. Check if the covariance matrix of your data is singular. For example, check for any pairs of variables with extreme correlations (> .95)."))
-      } else {
-        .quitAnalysis(gettextf("BDgraph failed with the following error message:\n%s", message))
+
+    # if model is ordinal Markov random field
+    if (options[["model"]] == "omrf") {
+      for (var in options[["variables"]]) {
+        # Check if variables are binary or ordinal:
+        if (!is.factor(dataset[[nw]][[var]])) {
+          .quitAnalysis(gettext("Some of the variables you have entered for analysis are not binary or ordinal. Please make sure that all variables are binary or ordinal or change the model to gcgm."))
+        } else {
+          # Estimate mixed network
+          jaspBase::.setSeedJASP(options)
+          easybgmFit <- try(easybgm::easybgm(data       = as.data.frame(dataset[[nw]]),
+                                             type       = "ordinal",
+                                             package    = "bgms",
+                                             iter       = as.numeric(options[["iter"]]),
+                                             save       = TRUE,
+                                             centrality = FALSE,
+                                             burnin     = as.numeric(options[["burnin"]]),
+                                             inclusion_probability = as.numeric(options[["gprior"]]),
+                                             interaction_scale = as.numeric(options[["interactionScale"]]),
+                                             edge_prior = as.character(options[["edgePrior"]]),
+                                             threshold_alpha = as.numeric(options[["threshold_alpha"]]),
+                                             threshold_beta = as.numeric(options[["threshold_beta"]]),
+                                             beta_bernoulli_alpha = as.numeric(options[["beta_alpha"]]),
+                                             beta_bernoulli_beta = as.numeric(options[["beta_beta"]]),
+                                             dirichlet_alpha = as.numeric(options[["dirichlet_alpha"]])))
+
+        }
       }
```

Comment:
Aiaiai, I knew the omrf was relatively slow, but this takes it to the next level! You're looping over the variables, running the omrf for each one, and discarding all results except for the last one! 

I would suggest the following changes.
If omrf is selected do *once* at the beginning of the function the error check on the data type. Then at this point you can assume that the data has the correct format and run the analysis.

Reply:
ohh yes!